src/i18n.ts

```diff
@@ -0,0 +1,122 @@
+import { readFileSync, readdirSync, existsSync } from "node:fs";
+import { join } from "node:path";
+
+let language: ioBroker.Languages = "en";
+let words: null | Record<string, ioBroker.Translated> = null;
+
+// Init internationalization
+export async function init(
+	/** The root directory of the adapter */
+	rootDir: string,
+	/** The adapter instance or the language to use */
+	langOrAdmin: ioBroker.Adapter | ioBroker.Languages,
+): Promise<void> {
+	if (langOrAdmin && typeof langOrAdmin === "object") {
+		const adapter = langOrAdmin as ioBroker.Adapter;
```

Comment:
the `as` type casting is not necessary TS can infer it from the check and the possible types